aidb/inference/inference_service.py

```diff
@@ -27,6 +27,7 @@ class InferenceService(abc.ABC):
   is_single: bool  # Return a single value or a list of values
   cost: Union[float, None] = None
   preferred_batch_size: int = 1
+  copied_input_columns: List[int] = field(default_factory=list)
```

Comment:
What is this? I think this needs to be documented

Reply:
This is to specify which columns should be copied. In infer_batch, one input may produce 0-inf outputs. It is too late if we do copying input after transforming json to pandas dataframes. Also, by having this, we can avoid binding a copied input column to multiple inference services because they will not be binded at all. Currently users need to specify the index to `input_col` that need to be copied to output. For example, if we have
```python
inference_engines = [
  {
    "service": ocr,
    "input_col": ("pdf.path", "pdf.id"),
    "output_col": ("ocr.text", "ocr.id")
  },
]
```
and we want to copy "pdf.id" to output, we specify `copied_input_columns=[1],`

This new feature wil be tested later this week, as I was working on optimization and debugging previously.

Reply:
This is fine but can we add documentation somewhere?

Reply:
yes, a doc is actually available at https://github.com/ddkang/aidb/tree/conrevo/infer_batch/config#inference-service-configuration

the corresponding pr change is at https://github.com/ddkang/aidb/pull/148/files#diff-6e8c7fdec58c9cb55af12d91063af268d43c753cf477bc41ec4e938f1b429b6f